core/services/vrf/listener_v2.go

```diff
@@ -571,12 +571,7 @@ func (lsn *listenerV2) processRequestsPerSubBatch(
 		}
 
 		fromAddresses := lsn.fromAddresses()
-		fromAddress, err := lsn.gethks.GetRoundRobinAddress(lsn.chainID, fromAddresses...)
-		if err != nil {
-			l.Errorw("Couldn't get next from address", "err", err)
-			continue
-		}
-		maxGasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddress)
+		maxGasPriceWei := lsn.cfg.KeySpecificMaxGasPriceWei(fromAddresses[0])
```

Comment:
Worth also adding a comment here stating that it is expected that all key specific max gas prices for all the from addresses should be the same, so it's okay to take just the first one's max gas price.